Formula/nginx-unit.rb

```diff
@@ -0,0 +1,256 @@
+      assert_equal "200", response.code
+      assert_equal "Hello, perl world!", response.body
+
+      ### Python module
+
+      socket = Net::BufferedIO.new(UNIXSocket.new(control))
+      request = Net::HTTP::Put.new("/config")
+      request.body = JSON.generate({})
+      request.exec(socket, "1.1", request.path)
+      response = Net::HTTPResponse.read_new(socket)
+      response.reading_body(socket, request.response_body_permitted?) {}
+
+      pythonapp = testpath/"python/wsgi.py"
+      pythonapp.write <<~EOS
+        def application(environ, start_response):
+          start_response('200 OK', [('Content-Type', 'text/plain')])
+          return ["Hello, python world!"]
+      EOS
+
+      socket = Net::BufferedIO.new(UNIXSocket.new(control))
+      request = Net::HTTP::Put.new("/config")
+      request.body = JSON.generate(
+        "listeners" => { "*:8000" => { "application" => "pythonapp" } },
+        "applications" => { "pythonapp" => { "type" => "python", "path" => pythonapp.parent, "module" => "wsgi" } },
+      )
+      request.exec(socket, "1.1", request.path)
+      response = Net::HTTPResponse.read_new(socket)
+      response.reading_body(socket, request.response_body_permitted?) {}
+
+      assert_equal "200", response.code
+      expected_body = { "success" => "Reconfiguration done." }
+      assert_equal expected_body, JSON.parse(response.body)
+
+      response = Net::HTTP.get_response(URI.parse("http://localhost:8000"))
+      assert_equal "200", response.code
+      assert_equal "Hello, python world!", response.body
+
+      ### Ruby module
+
+      socket = Net::BufferedIO.new(UNIXSocket.new(control))
+      request = Net::HTTP::Put.new("/config")
+      request.body = JSON.generate({})
+      request.exec(socket, "1.1", request.path)
+      response = Net::HTTPResponse.read_new(socket)
+      response.reading_body(socket, request.response_body_permitted?) {}
+
+      rubyapp = testpath/"ruby/config.ru"
+      rubyapp.write <<~EOS
+        run lambda { |env| [200, {"Content-Type" => "text/plain"}, ["Hello, ruby world!"]]}
+      EOS
+
+      socket = Net::BufferedIO.new(UNIXSocket.new(control))
+      request = Net::HTTP::Put.new("/config")
+      request.body = JSON.generate(
+        "listeners" => { "*:8000" => { "application" => "rubyapp" } },
+        "applications" => { "rubyapp" => { "type" => "ruby", "working_directory" => rubyapp.parent, "script" => rubyapp } },
+      )
+      request.exec(socket, "1.1", request.path)
+      response = Net::HTTPResponse.read_new(socket)
+      response.reading_body(socket, request.response_body_permitted?) {}
+
+      assert_equal "200", response.code
+      expected_body = { "success" => "Reconfiguration done." }
+      assert_equal expected_body, JSON.parse(response.body)
+
+      response = Net::HTTP.get_response(URI.parse("http://localhost:8000"))
+      assert_equal "200", response.code
+      assert_equal "Hello, ruby world!", response.body
+    rescue
```

Comment:
We shouldn't really be rescuing in a test block.

Reply:
It's impossible to debug the tests without the log output. Is there another way you recommend capturing log info for debugging tests?

Reply:
Add a `raise "here"` to the test. And then run `brew test --debug`

Reply:
That doesn't work for CI, though?

Reply:
If it fails on CI, it needs to be debugged locally.

Reply:
It passes locally. It only fails on CI.

Reply:
OK, feel free to debug that with the rescue but let's make sure it doesn't get merged with that in there.